### PR TITLE
perf: query accounts with `&Address` to avoid copying address

### PR DESCRIPTION
**Pull request by hai-rise**
Following #13171 towards #12838.

### CODE REVIEW

File: crates/transaction-pool/src/validate/eth.rs [hunk 1]
```diff
@@ -367,7 +367,7 @@ where
         let state = maybe_state.as_deref().expect("provider is set");
 
         // Use provider to get account info
-        let account = match state.basic_account(transaction.sender()) {
+        let account = match state.basic_account(transaction.sender_ref()) {
```
Comment (hai-rise) [inline crates/transaction-pool/src/validate/eth.rs:370]: Main dance.

File: crates/storage/provider/src/providers/database/provider.rs [hunk 2]
```diff
@@ -862,8 +862,8 @@ impl<TX: DbTxMut + DbTx + 'static, N: NodeTypes> DatabaseProvider<TX, N> {
 }
 
 impl<TX: DbTx, N: NodeTypes> AccountReader for DatabaseProvider<TX, N> {
-    fn basic_account(&self, address: Address) -> ProviderResult<Option<Account>> {
-        Ok(self.tx.get::<tables::PlainAccountState>(address)?)
+    fn basic_account(&self, address: &Address) -> ProviderResult<Option<Account>> {
+        Ok(self.tx.get_by_encoded_key::<tables::PlainAccountState>(address)?)
```
Comment (hai-rise) [inline crates/storage/provider/src/providers/database/provider.rs:866]: Main dance.

File: crates/storage/provider/src/providers/state/latest.rs [hunk 3]
```diff
@@ -43,8 +43,8 @@ impl<'b, Provider: DBProvider> LatestStateProviderRef<'b, Provider> {
 
 impl<Provider: DBProvider> AccountReader for LatestStateProviderRef<'_, Provider> {
     /// Get basic account information.
-    fn basic_account(&self, address: Address) -> ProviderResult<Option<Account>> {
-        self.tx().get::<tables::PlainAccountState>(address).map_err(Into::into)
+    fn basic_account(&self, address: &Address) -> ProviderResult<Option<Account>> {
+        self.tx().get_by_encoded_key::<tables::PlainAccountState>(address).map_err(Into::into)
```
Comment (hai-rise) [inline crates/storage/provider/src/providers/state/latest.rs:47]: Main dance.

File: crates/storage/provider/src/providers/database/provider.rs (hunk 2 above)
Comment (mattsse) [reply crates/storage/provider/src/providers/database/provider.rs:866]: makes sense to avoid the copy here

**Review by mattsse — APPROVED** [review]
cool, this makes sense

**Pull request merged**